File: differ/src/main/java/ru/starksoft/differ/viewholder/ClickAction.java
package ru.starksoft.differ.viewholder;

import androidx.annotation.NonNull;
import ru.starksoft.differ.utils.hash.Cache;

public final class ClickAction {

	private static final Cache cache = new Cache();

	private ClickAction() {
		throw new UnsupportedOperationException();
	}

	public static int getActionId(@NonNull Class clazz, @NonNull String name) {
		return cache.getValue(clazz.getName() + ":" + name);
	}
}

File: diff-adapter/src/main/java/ru/starksoft/differ/adapter/viewholder/ClickAction.kt
package ru.starksoft.differ.adapter.viewholder

import ru.starksoft.differ.utils.hash.Cache

object ClickAction {

    private val cache by lazy { Cache() }

    fun getActionId(clazz: Class<*>, name: String): Int {
        return cache["${clazz.name}:$name"]
    }

    operator fun get(clazz: Class<*>, name: String): Int {
        return cache["${clazz.name}:$name"]
    }
}
